src/Avalonia.ReactiveUI/ReactiveUserControl.cs

```diff
@@ -20,7 +25,13 @@ public class ReactiveUserControl<TViewModel> : UserControl, IViewFor<TViewModel>
         /// </summary>
         public ReactiveUserControl()
         {
-            DataContextChanged += (sender, args) => ViewModel = DataContext as TViewModel;
+            this.WhenActivated(disposables => { });
+            this.WhenAnyValue(x => x.ViewModel)
+                .Skip(1)
+                .Subscribe(model => DataContext = model);
+            this.WhenAnyValue(x => x.DataContext)
```

Comment:
You can use `ObeservableForProperty` here instead. That way you don't need to `Skip`

Reply:
Good catch, thanks! Going to correct this soon.